### PR TITLE
[1.4.1] Set webhook matchPolicy to Exact (#4271)

### PR DESCRIPTION
**Pull request by charith-elastic**
Backport of #4271

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,7 +1,5 @@
-cloud.google.com/go v0.26.0 h1:e0WKqKTd5BnrG8aKH3J3h+QvEIQtSUcf2n5UZ5ZgLtQ=
```
Comment (thbkrkr) [inline go.sum:1]: Why is this file modified?

Comment (charith-elastic) [reply go.sum:1]: Go 1.16 gets rid of those entries when you run `go mod tidy`. I don't know the reason why. The changes seem to be harmless given that the E2E tests ran successfully so I kept them on.

**Review by thbkrkr — APPROVED** [review]
LGTM
